WordPress/src/main/java/org/wordpress/android/ui/jetpack/JetpackCapabilitiesUseCase.kt

```diff
@@ -31,58 +30,62 @@ class JetpackCapabilitiesUseCase @Inject constructor(
-            val lastUpdated = appPrefsWrapper.getSiteJetpackCapabilitiesLastUpdated(remoteSiteId)
-            lastUpdated > currentDateProvider.currentDate().time - MAX_CACHE_VALIDITY
-        }
+    suspend fun fetchJetpackPurchasedProducts(remoteSiteId: Long): JetpackPurchasedProducts =
+            mapToJetpackPurchasedProducts(fetchJetpackCapabilities(remoteSiteId))
+
+    private fun mapToJetpackPurchasedProducts(capabilities: List<JetpackCapability>) =
+            JetpackPurchasedProducts(
+                    scan = capabilities.find { SCAN_CAPABILITIES.contains(it) } != null,
+                    backup = capabilities.find { BACKUP_CAPABILITIES.contains(it) } != null
+            )
+
+    fun hasValidCache(remoteSiteId: Long): Boolean {
+        val lastUpdated = appPrefsWrapper.getSiteJetpackCapabilitiesLastUpdated(remoteSiteId)
+        return lastUpdated > currentDateProvider.currentDate().time - MAX_CACHE_VALIDITY
     }
 
-    private suspend fun getCachedJetpackCapabilities(remoteSiteId: Long): List<JetpackCapability> {
-        return withContext(bgDispatcher) { appPrefsWrapper.getSiteJetpackCapabilities(remoteSiteId) }
+    private fun getCachedJetpackCapabilities(remoteSiteId: Long): List<JetpackCapability> {
+        return appPrefsWrapper.getSiteJetpackCapabilities(remoteSiteId)
     }
 
     private suspend fun fetchJetpackCapabilities(remoteSiteId: Long): List<JetpackCapability> {
-        return withContext(bgDispatcher) {
-            if (continuation != null) {
-                throw IllegalStateException("Request already in progress.")
-            }
-
-            dispatcher.register(this@JetpackCapabilitiesUseCase)
-            val response = suspendCoroutine<OnJetpackCapabilitiesFetched> { cont ->
-                val payload = FetchJetpackCapabilitiesPayload(remoteSiteId)
-                continuation = cont
-                dispatcher.dispatch(SiteActionBuilder.newFetchJetpackCapabilitiesAction(payload))
-            }
-
-            val capabilities: List<JetpackCapability> = response.capabilities ?: listOf()
-            if (!response.isError) {
-                updateCache(remoteSiteId, capabilities)
-            }
-            return@withContext capabilities
+        forceResumeDuplicateRequests(remoteSiteId)
+
+        dispatcher.register(this@JetpackCapabilitiesUseCase)
+        val response = suspendCoroutine<OnJetpackCapabilitiesFetched> { cont ->
+            val payload = FetchJetpackCapabilitiesPayload(remoteSiteId)
+            continuation[remoteSiteId] = cont
+            dispatcher.dispatch(SiteActionBuilder.newFetchJetpackCapabilitiesAction(payload))
+        }
+
+        val capabilities: List<JetpackCapability> = response.capabilities ?: listOf()
+        if (!response.isError) {
+            updateCache(remoteSiteId, capabilities)
+        }
+        return capabilities
+    }
+
+    private fun forceResumeDuplicateRequests(remoteSiteId: Long) {
+        continuation[remoteSiteId]?.let {
```

Comment:
Instead of throwing an exception, we simply resume any duplicate requests with an exception.